### PR TITLE
JAMES-1648 Handle header filtering

### PR DESCRIPTION
**Pull request by rouazana**

### CODE REVIEW

File: server/protocols/jmap/pom.xml [hunk 1]
```diff
@@ -243,6 +243,12 @@
                     <type>test-jar</type>
                 </dependency>
                 <dependency>
+                    <groupId>com.jayway.jsonpath</groupId>
```
Comment (aduprat) [inline server/protocols/jmap/pom.xml]: Version & artifact may be put in parent as this artifact is shared with more than one project.

Comment (rouazana) [inline server/protocols/jmap/pom.xml]: ok

File: server/pom.xml [hunk 2]
```diff
@@ -1394,6 +1394,11 @@
                 <artifactId>awaitility</artifactId>
                 <version>1.6.3</version>
             </dependency>
+                <dependency>
```
Comment (mbaechler) [inline server/pom.xml]: ah ah, indent !

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/JmapResponseWriterImpl.java [hunk 3]
```diff
@@ -59,20 +60,34 @@ public JmapResponseWriterImpl(Set<Module> jacksonModules) {
...
     
     private ObjectMapper newConfiguredObjectMapper(JmapResponse jmapResponse) {
-        ObjectMapper objectMapper = new ObjectMapper().registerModules(jacksonModules)
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.registerModules(jacksonModules)
                 .configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false);
-
-        objectMapper.setFilterProvider(buildPropertiesFilter(jmapResponse.getProperties()));
+        
+        FilterProvider filterProvider = jmapResponse
```
Comment (mbaechler) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/JmapResponseWriterImpl.java]: It's just a detail, but you could write : 
```
jmapResponse.getFilterProvider().orElseGet(new SimpleFilterProvider()).addFilter(blabla)
```

File: server/pom.xml (hunk 2 above)
Comment (rouazana) [inline server/pom.xml]: ok

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/JmapResponseWriterImpl.java (hunk 3 above)
Comment (rouazana) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/JmapResponseWriterImpl.java]: ok, with some casting added

**Comment by mbaechler** [comment]
:+1:

**Comment by aduprat** [comment]
:+1:

**Pull request closed without merging**